Source/WebGPU/WebGPU/Texture.mm

```diff
@@ -1961,20 +1961,32 @@ static MTLPixelFormat pixelFormat(WGPUTextureFormat textureFormat)
     }
 }
 
-static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, bool supportsNonPrivateDepthStencilTextures)
+static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, bool supportsNonPrivateDepthStencilTextures, bool isBackedByIOSurface)
 {
+    // IOSurface-backed texture must be MTLStorageModeManaged, else the driver will assert with
+    // "Texture Descriptor Validation IOSurface textures must use MTLStorageModeManaged"
```

Comment:
I think this second clause is unnecessary. I'd rewrite the comment to say "Metal validation requires IOSurface-backed textures to be MTLStorageModeManaged."

Is this comment true for devices with unified memory? If not, we should probably change the logic in this function.

Reply:
(See below comment about `isForDepthOrStencil`) Can we add something here like

```
ASSERT(!isForDepthOrStencil || !isBackedByIOSurface);
```

Reply:
> Is this comment true for devices with unified memory? If not, we should probably change the logic in this function.

I hit the assertion on a a M1 MacBook Pro so yes